### PR TITLE
Refactor solid-client to use component libraries

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Extract authentication, permissions, web-client and vocab into their own standalone
libraries. Use rdflib and webClient as dependency injections.

Partly addresses issue #30.

The following modules have been extracted to standalone repos:
* [`solid-web-client`](https://github.com/solid/solid-web-client)
* [`solid-permissions`](https://github.com/solid/solid-permissions)
* [`solid-auth-tls`](https://github.com/solid/solid-auth-tls)
* [`solid-namespace`](https://github.com/solid/solid-namespace)

### CODE REVIEW

**Comment by dan-f** [comment]
I'm wondering what the difference between the `full` and the `with-rdflib` builds are.

**Comment by dan-f** [comment]
I'd recommend having a single webpack config if possible (so that settings don't get out of sync). I think that all (or nearly all) of the webpack settings can also be specified by command-line, and I wonder if the CLI options take precedence.

If that doesn't work, another good way to go is to have a 'base' webpack config and then have your other webpack configs import and override settings, since webpack configs are just regular CommonJS modules.

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 language: node_js
 node_js:
-  - "5.0"
+  - "6.0"
```
Comment (dan-f) [inline .travis.yml]: Could you document the supported node versions in the `README.md` and `CHANGELOG.md`?

Comment (dmitrizagidulin) [inline .travis.yml]: Will do. Thanks!

File: index.js [hunk 2]
```diff
@@ -33,29 +33,46 @@ https://github.com/solid/solid
...
   AppRegistration: require('./lib/solid/app-registration'),
   appRegistry: require('./lib/app-registry'),
-  auth: require('./lib/auth'),
+  auth: auth,
   config: require('./config'),
-  currentUser: require('./lib/auth').currentUser,
-  getProfile: require('./lib/identity').getProfile,
-  getPermissions: require('./lib/permissions').getPermissions,
-  clearPermissions: require('./lib/permissions').clearPermissions,
+  currentUser: auth.currentUser,
```
Comment (dan-f) [inline index.js]: Can you confirm a backwards-incompatible API change here? Looks like calls to `solid.getProfile`, for example, will have to update to `solid.auth.getProfile`? Could you add that to the release notes if so?

Comment (dan-f) [inline index.js]: Whoops I should have read down further.

**Comment by dan-f** [comment]
Really great work!!

:+1: pending what we've discussed (if you want to follow-up PR, that's cool!).

**Comment by dmitrizagidulin** [comment]
Refactored webpack config. Merging.

**Pull request merged**